node/network/availability-distribution/src/tests.rs

```diff
@@ -1,1278 +0,0 @@
-	let para_b = ParaId::from(2);
-	let para_c = ParaId::from(3);
-
-	let make_candidate = |para_id| {
-		let mut candidate = CommittedCandidateReceipt::default();
-		candidate.descriptor.para_id = para_id;
-		candidate.descriptor.relay_parent = [69u8; 32].into();
-		candidate
-	};
-
-	let candidate_a = make_candidate(para_a);
-	let candidate_b = make_candidate(para_b);
-	let candidate_c = make_candidate(para_c);
-
-	let candidate_hash_a = candidate_a.hash();
-	let candidate_hash_b = candidate_b.hash();
-	let candidate_hash_c = candidate_c.hash();
-
-	// receipts has an initial entry for hash_a but not hash_b.
-	let mut receipts = HashMap::new();
-	receipts.insert(hash_a, vec![candidate_hash_a, candidate_hash_b].into_iter().collect());
-
-	let pool = sp_core::testing::TaskExecutor::new();
-
-	let (mut ctx, mut virtual_overseer) =
-		test_helpers::make_subsystem_context::<AvailabilityDistributionMessage, _>(pool);
-
-	let test_fut = async move {
-		let live_candidates = query_pending_availability_at(
-			&mut ctx,
-			vec![hash_a, hash_b],
-			&mut receipts,
-		).await.unwrap();
-
-		// although 'b' is cached from the perspective of hash_a, it gets overwritten when we query what's happening in
-		//
-		assert_eq!(live_candidates.len(), 3);
-		assert_matches!(live_candidates.get(&candidate_hash_a).unwrap(), FetchedLiveCandidate::Cached);
-		assert_matches!(live_candidates.get(&candidate_hash_b).unwrap(), FetchedLiveCandidate::Cached);
-		assert_matches!(live_candidates.get(&candidate_hash_c).unwrap(), FetchedLiveCandidate::Fresh(_));
-
-		assert!(receipts.get(&hash_b).unwrap().contains(&candidate_hash_b));
-		assert!(receipts.get(&hash_b).unwrap().contains(&candidate_hash_c));
-	};
-
-	let answer = async move {
-		// hash_a should be answered out of cache, so we should just have
-		// queried for hash_b.
-		assert_matches!(
-			overseer_recv(&mut virtual_overseer).await,
-			AllMessages::RuntimeApi(
-				RuntimeApiMessage::Request(
-					r,
-					RuntimeApiRequest::AvailabilityCores(tx),
-				)
-			) if r == hash_b => {
-				let _ = tx.send(Ok(vec![
-					CoreState::Occupied(OccupiedCore {
-						next_up_on_available: None,
-						occupied_since: 0,
-						time_out_at: 0,
-						next_up_on_time_out: None,
-						availability: Default::default(),
-						group_responsible: GroupIndex::from(0),
-						candidate_hash: candidate_hash_b,
-						candidate_descriptor: candidate_b.descriptor.clone(),
-					}),
-					CoreState::Occupied(OccupiedCore {
-						next_up_on_available: None,
-						occupied_since: 0,
-						time_out_at: 0,
-						next_up_on_time_out: None,
-						availability: Default::default(),
-						group_responsible: GroupIndex::from(0),
-						candidate_hash: candidate_hash_c,
-						candidate_descriptor: candidate_c.descriptor.clone(),
-					}),
-				]));
-			}
-		);
-	};
-
-	futures::pin_mut!(test_fut);
-	futures::pin_mut!(answer);
-
-	executor::block_on(future::join(test_fut, answer));
-}
-
-#[test]
-fn new_peer_gets_all_chunks_send() {
```

Comment:
it seems that more tests were removed than added, is that intentional? (haven't looked deep enough)

Reply:
The removed tests, are dead tests of the old implementation. Hardly comparable to the new implementation. Also the added tests are in addition to tests that have been already present in the original PR.